File: port/PlayerList.py
import Consts
from Player import Player
from HumanPlayer import HumanPlayer
from NetworkPlayer import NetworkPlayer
from AI_Sample_MaxActEval import AI_Sample_MaxActEval
from AI_MSSystem import AI_MSSystem
from AI_M_UCT import AI_M_UCT
from AI_M3Lee_hull import AI_M3Lee_hull
from AI_SatD2 import AI_SatD2

class PlayerList:
    """
    Class for registering player types.
    Please edit this when you add a new thinking routine.
    """

    # List of player objects. Please register the necessary classes here.
    registered_player_list = [
        HumanPlayer(),  # Don't change this
        NetworkPlayer(),  # Don't change this
        AI_Sample_MaxActEval(),
        AI_MSSystem(),
        AI_M_UCT(),
        AI_M3Lee_hull(),
        AI_SatD2()
    ]

    # Register the indices of the 2 players you want to compete by default here.
    default_player_index = [0, 2]

    # Constants for player types
    HUMAN_PLAYER = 0
    NETWORK_PLAYER = 1

    @staticmethod
    def get_player(player_index):
        """
        Return the player object specified by the index.
        :param player_index: Index of the player
        :return: Player object
        """
        return PlayerList.registered_player_list[player_index]

    @staticmethod
    def initialize_player_list_combo_box(cb_red, cb_blue):
        """
        Initialize the player list combo box.
        :param cb_red: Combo box for the red team
        :param cb_blue: Combo box for the blue team
        """
        cb_red.clear()
        cb_blue.clear()
        for player in PlayerList.registered_player_list:
            cb_red.addItem(player.get_name())
            cb_blue.addItem(player.get_name())
        cb_red.setCurrentIndex(PlayerList.default_player_index[Consts.RED_TEAM])
        cb_blue.setCurrentIndex(PlayerList.default_player_index[Consts.BLUE_TEAM])

File: PlayerList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace SimpleWars {
    // Playerの種類を登録するためのクラス
    // 新しい思考ルーチンを追加した場合はここを編集してください．
    class PlayerList {

        // プレイヤーオブジェクトのリスト．ここに必要なクラスを登録してください
        public static Player[] sRegisteredPlayerList = new Player[] {
            new HumanPlayer(),           // これは変更しないでください
            new NetworkPlayer(),         // これは変更しないでください
			new AI_M_UCT(),
            new AI_Sample_MaxActEval(),
            new AI_RHEA(),
            new AI_RHCP(),
            new AI_TBETS(),
            new AI_EMCTS(),
            new AI_EMCTS_FH(),
            // new AI_SatTS_D2(),
            new AI_M_UCT_PW()
        };

        // デフォルトで対戦させたい 2プレイヤーのインデックスをここに登録する
        public static readonly int[] sDefaultPlayerIndex = new int[2] { 0, 2 };

        // 以降はユーザがいじらなくてもよい部分 -------------

        // playerの種類
        public const int HUMAN_PLAYER = 0;
        public const int NETWORK_PLAYER = 1;

        // indexに指定されたPlayerオブジェクトを返す
        public static Player getPlayer(int playerIndex) {
            return sRegisteredPlayerList[playerIndex];
        }

        // コンボボックスのプレイヤーリストを初期化する
        public static void initializePlayerListComboBox(ComboBox cb_red, ComboBox cb_blue) {
            cb_red.Items.Clear();
            cb_blue.Items.Clear();

            foreach (Player p in sRegisteredPlayerList) {
                cb_red.Items.Add(p.getName());
                cb_blue.Items.Add(p.getName());
            }

            cb_red.SelectedIndex = sDefaultPlayerIndex[Consts.RED_TEAM];
            cb_blue.SelectedIndex = sDefaultPlayerIndex[Consts.BLUE_TEAM];
        }
    }
}
